Issue#33 find inactive customers and test

adamgobhardter opened the pull request:
Implemented findInactiveCustomers method and created a mock test for it. I tried to do it in a TDD way, but since the method is not that complex only one cycle (test, code, refactor) happened.

Closes #33

Commit:
Update CustomerServiceMockitoTest.java

Created a mock test for findInactiveCustomers method and modified test file in some places. The findInactiveCustomers method will be implemented in the future.

Commit:
Implemented findInactiveCustomers method

BAndriss commented on src/main/java/hu/bme/aut/temalab/temalaborvehicleleasing/service/CustomerService.java:
```diff
@@ -73,4 +77,27 @@ public void giveOutBonusPoints(int amount, int topHowMany) {
+			rentals = List.copyOf(rentalRepository.findByCustomer(c));
+			
+			if(rentals != null && !rentals.isEmpty()) {
+				if(ChronoUnit.YEARS.between(rentals.get(rentals.size() - 1).getStartDate(), LocalDate.now()) > 2) inactiveCustomers.add(c);
```
A rentals rendezve van időpont alapján vagy honnét tudjuk, hogy az utolsó rental az?

adamgobhardter replied:
Hát ugye úgy van, hogy ha egy customer kibérel egy járművet, akkor a rentals tömbjébe bekerül egy rental. Ezáltal rentals tömb a startDate alapján rendezett lesz (növekvő).

BAndriss replied:
Igen, de akkor figyelni kell rá, hogy a rentals több milyen sorendbe kerülnek be az adatok. Pl.: Valaki plusz adatott kézzel bele rak, aminek korábbi a dátuma és már nincs benne a 2 évben akkor már inaktív lesz az adot customer

BAndriss replied:
Pl..: ha kipróbáloda mockito tesztnél, hogy az utolsó 2 rentals.add-ot megcseréled már elrontja a tesztett... Mivel ott hiába nem régi a bérlés nem azt nézi hanem a régebbit

BAndriss replied:
Yap, ez is igaz, igazából szerintem maradhat így max ha gond lesz vele később módosítunk rajta

Commit:
Fixed findInactiveCustomers method

Fixed a bug (suggested by @BAndriss ) in findInactiveCustomers method and refined findInactiveCustomers test.

Closes #33

BAndriss reviewed (approved):
Yap így már egyértelmű
